app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt

```diff
@@ -13,17 +14,24 @@ import com.karhoo.samples.uisdk.dropin.config.AdyenTokenExchangeConfig
 import com.karhoo.samples.uisdk.dropin.config.BraintreeGuestConfig
 import com.karhoo.samples.uisdk.dropin.config.BraintreeTokenExchangeConfig
 import com.karhoo.sdk.api.KarhooApi
-import com.karhoo.sdk.api.KarhooApi.setConfiguration
 import com.karhoo.sdk.api.KarhooError
 import com.karhoo.sdk.api.network.request.UserLogin
 import com.karhoo.sdk.api.network.response.Resource
 import com.karhoo.uisdk.KarhooUISDK
 import com.karhoo.uisdk.screen.booking.BookingActivity
+import kotlin.system.exitProcess
 
 class MainActivity : AppCompatActivity() {
 
     private lateinit var loadingProgressBar: View
 
+    init {
+        Thread.setDefaultUncaughtExceptionHandler { _, _ ->
+            this.startActivity(Intent(this, MainActivity::class.java))
+            exitProcess(0)
```

Comment:
Would it be better to clear the activity stack and start the MainActivity or SplashActivity? 
e.g. https://tips.androidhive.info/2013/10/how-to-clear-all-activity-stack-in-android/

Reply:
When getting an unchecked error, the application might enter some strange state. Exiting the process seems like the way to go in order to be certain that the app will enter a fresh state when restarting the main activity